src/vsphere_cpi/lib/cloud/vsphere/vm_creator.rb

```diff
@@ -261,15 +261,16 @@ def create(vm_config)
               @client.reconfig_vm(created_vm_mob, config_spec)
             end
             unless vm_config.pci_passthroughs.empty?
-              config_spec = VimSdk::Vim::Vm::ConfigSpec.new
               vm_config.pci_passthroughs.each do |pci_passthrough|
                 virtual_pci_passthrough = Resources::PCIPassthrough.create_pci_passthrough(
                   vendor_id: pci_passthrough['vendor_id'],
                   device_id: pci_passthrough['device_id'],
                 )
+                config_spec = VimSdk::Vim::Vm::ConfigSpec.new
                 config_spec.device_change << Resources::VM.create_add_device_spec(virtual_pci_passthrough)
```

Comment:
@lnguyen @adam-jian-zhang 

When I update a VM to have one GPU, BOSH will delete this VM and recreate a VM on a ESX that has GPU resources with attaching one GPU to it.
So I feel VC or somewhere seems having some scheduling ability.

 If VC has the ability to find out available ESX having enough GPU (not sure, you're both the experts :) ),  I am wondering if it has some difference between adding device multiple times to a VM with adding multiple devices in one time to a VM, and if the latter approach can increase the success rate of vm creation.